### ISSUE TITLE
Moderate warnings from ACE for sections/asides with no title, or a title that repeats

### ISSUE THREAD

**Issue by AndersEkl** [nlbdev/nordic-accessible-epub-guidelines#39]
Originally written by @AndersEkl 

ACE give out moderate warnings for `<aside>` elements without `aria-label` or `aria-labelledby` as soon as there are more than one `<aside>` element in one content file. How do we handle this? There could be hundreds of `<aside>` elements in one EPUB file.

Comment by @martinpub 

This is similar to untitled `<section>` elements, right? I suggest something very simple, such as "Box 1", "Box 2" (localised), to comply with the ARIA requirements. It could easily be automated on the producers' side, based on the `@id` values, just in a (slightly more) readable form.

Another case (MTM production V006427, for the record) where the same ACE validation error is triggered: Multiple subsection headings with the same heading text (below different parent headings/sections, in the same content file). Markup is done according to the print source, so it seems accessibility production and truthful replication of original content are at odds here.

To clarify, the headings/sections are discernible from each other on context (nesting, parent headings/section, etc.), but the actual label is the same, and according to  the ACE/aXe rule `landmarks-unique`, this is an accessibility issue.

**Comment by AndersEkl** [comment]
I have been thinking about this and maybe it stems from us using `<aside>` in a wrong way? Let's discuss this further when revision work starts.

**Comment by jonaslil** [comment]
The warning from Ace is moderate and concerns a best practice rule, so it's not the most serious issue, but still worth disussing.

The case with subsections triggering this error is connected to #2 and should be solved if we restrict the use of labelledby to the most important sections.

**Comment by martinpub** [comment]
Thanks for your input @jonaslil, and yes I agree.

**Comment by AndersEkl** [comment]
Yes, I agree, @jonaslil . That was our reasoning in the 2020-1 version. I think we should look into minimising the volume of warnings, but if we can't eliminate them altogether (probably not possible) we should mention in the guidelines that these warnings will occur.

**Comment by martinpub** [comment]
Do we agree that currently, we are only mentioning that multiple asides in content documents trigger warnings/make asides harder do disambiguate in AT?

See also http://kb.daisy.org/publishing/docs/html/asides.html#faq-003.

**Comment by jonaslil** [comment]
I agree. Also noting this on the KB page about asides:

> The default nature of the aside element is changing. asides without an accessible name that are within a section element will no longer be considered complementary landmarks. They will instead be considered generic elements (i.e., the same as div elements).

Maybe this means that ACE will stop complaining about asides without accessible names at some point.

**Comment by martinpub** [comment]
> Maybe this means that ACE will stop complaining about asides without accessible names at some point.

Oh, yes I didn't see that. Then I believe this should in the future have ACE stop complaining, yes. Good times.

**Comment by martinpub** [comment]
TODO Martin: Write a note about the warnings, and refer to the WAI-ARIA change of asides.

**Comment by martinpub** [comment]
Hi @AndersEkl @jonaslil @oscarlcarlsson and @lindheli,

I did some research on this to prepare the text. It seems the current specification [HTML Accessibility API Mappings 1.0 already reflects this](https://www.w3.org/TR/html-aam-1.0/#el-aside), that asides only are landmarks if they have an accessible name.

**Comment by martinpub** [comment]
Hope the text is understandable? Closing for now.

**Comment by martinpub** [comment]
Hi

**Comment by AndersEkl** [comment]
Yes, sounds reasonable.

**Comment by martinpub** [comment]
Thanks for confirming @AndersEkl. I will close this now, but feel free to reopen if you disagree Linda and Jonas.
